core/src/com/unciv/logic/civilization/ReligionManager.kt

```diff
@@ -227,36 +226,53 @@ class ReligionManager : IsPartOfGameInfoSerialization {
     }
 
     fun chooseBeliefs(iconName: String?, religionName: String?, beliefs: List<Belief>) {
-        when(religionState) {
+        when (religionState) {
             ReligionState.FoundingReligion ->
-                foundReligion(iconName!!, religionName!!, beliefs)
+                foundReligion(iconName!!, religionName!!)
             ReligionState.EnhancingReligion ->
-                enhanceReligion(beliefs)
-            else -> return
+                religionState = ReligionState.EnhancedReligion
+            ReligionState.None -> {
+                val belief = beliefs[0] // can only have one pantheon
+                choosePantheonBelief(belief)
+            }
+            else -> {}
+        }
+        // add beliefs (religion exists at this point)
+        religion!!.followerBeliefs.addAll(
+            beliefs
+                .filter { it.type == BeliefType.Pantheon || it.type == BeliefType.Follower }
+                .map { it.name }
+        )
+        religion!!.founderBeliefs.addAll(
+            beliefs
+                .filter { it.type == BeliefType.Founder || it.type == BeliefType.Enhancer }
+                .map { it.name }
+        )
+        // decrement free beliefs if used
+        if (freeBeliefs.sumValues() > 0) {
+            for (belief in beliefs) {
+                if (freeBeliefs[belief.type] == null) continue
+                if (religionState == ReligionState.Pantheon && belief.type == BeliefType.Pantheon
+                        && religion!!.followerBeliefs.size == 1 && storedFaith >= faithForPantheon(-1)) {
+                    // paid for the initial pantheon using faith, otherwise this civ is using a free pantheon belief pick
+                    storedFaith -= faithForPantheon(-1) // this civ is included in this calculation now so offset for that
+                    continue
+                }
+                freeBeliefs[belief.type] = max(freeBeliefs[belief.type]!! - 1, 0)
+            }
```

Comment:
I'm not entirely certain what's going on here. The decrementing of free beliefs when used is clear, but when exactly are you using the Faith cost? It looks like for that you already need to have a pantheon, and so also buy the second pantheon belief using Faith? Is that what you meant?

Reply:
The idea is that there's an edge case where the player can reach the amount of faith to form a pantheon on the same turn it builds a wonder/completes a tech that gives it a free pantheon pick and we want the player to have both pantheon belief picks. `chooseBeliefs()` is called first when the player/AI forms a pantheon and in this case we want to prioritize spending faith so the player can still choose the free belief. I do agree that these lines are confusing  as is, and with regards to your above comment, I've moved the faith decrease line back to `foundPantheon()` since we'll only be checking if the player has enough faith on the pantheon founding and not on any subsequent free pantheon picks.

Reply:
Ah I see. Wouldn't it in that case be easier to check at the start of this function (before the `when` statement) if you don't have a religion yet and then set a boolean flag, which can be read in the if statement?

Reply:
I'm doing something like that except giving `chooseBeliefs()` an optional boolean flag that indicates whether the player is using the free picks at this time since this case could also happen when a religion is formed/enhanced so just checking the faith amount alone isn't sufficient in the case where you're adding another Founder belief.